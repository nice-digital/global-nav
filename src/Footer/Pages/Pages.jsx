import React from "react";
import PropTypes from "prop-types";
import styles from "./Pages.module.scss";

import pages from "./links.json";
import TrackedLink from "../../TrackedLink";
import { footerClickEventAction } from "../../tracker";

//TODO add string type for service prop
const Pages = ({ service }) => {
	return (
		<nav className={styles.wrapper}>
			<ul className={styles.list}>
				{pages.map((link) => {
					let href = link.href;
					if (link.hrefOverrides && service) {
						href = link.hrefOverrides[service] || href;
					}
					return (
						<li key={link.text}>
							<TrackedLink
								href={href}
								eventAction={footerClickEventAction}
								eventLabel={link.text}
							>
								{link.text}
							</TrackedLink>
						</li>
					);
				})}
			</ul>
		</nav>
	);
};

Pages.propTypes = {
	service: PropTypes.string,
};

export default Pages;
