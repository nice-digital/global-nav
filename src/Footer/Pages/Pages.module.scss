@use 'sass:math';
@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';

.wrapper {
  margin: utils.rem(0 0 spacing.$large);
}

.list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: utils.rem(0 0 spacing.$x-small);
  }
}

@include media-queries.mq($from: xs) {
  .list {
    column-count: 2;
    column-gap: utils.rem(spacing.$medium);
  }
}

@include media-queries.mq($from: sm) {
  .wrapper {
    float: left;
    margin: 0;
    padding: utils.rem(0 0 0 spacing.$large);
    width: percentage(math.div(2, 3));
  }
}

@include media-queries.mq($from: md) {
  .wrapper {
    float: left;
    margin: 0;
    padding: utils.rem(0 spacing.$large);
    width: 50%;
  }

  .list {
    column-count: 2;
    column-gap: utils.rem(spacing.$medium);
  }
}
