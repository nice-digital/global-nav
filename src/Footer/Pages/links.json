[
  {
    "href": "https://www.nice.org.uk/what-nice-does",
    "text": "What NICE does"
  },
  {
    "href": "https://www.nice.org.uk/implementing-nice-guidance",
    "text": "Implementing NICE guidance"
  },
  {
    "href": "https://www.nice.org.uk/get-involved",
    "text": "Get involved"
  },
  {
    "href": "https://www.nice.org.uk/about-us",
    "text": "About us"
  },
  {
    "href": "https://www.nice.org.uk/careers",
    "text": "Careers"
  },
  {
    "href": "https://www.nice.org.uk/news",
    "text": "News, blogs and podcasts"
  },
  {
    "href": "https://www.nice.org.uk/nice-newsletters-and-alerts",
    "text": "Newsletters and alerts"
  },
  {
    "href": "https://www.nice.org.uk/contact-us",
    "text": "Contact us"
  },
  {
    "href": "https://www.nice.org.uk/forms/leave-feedback",
    "text": "Leave feedback"
  },
  {
    "href": "https://www.nice.org.uk/reusing-our-content",
    "text": "Reusing our content"
  },
  {
    "href": "https://www.nice.org.uk/forms/use-of-nice-content-in-the-uk",
    "text": "NICE UK Open Content Licence"
  },
  {
    "href": "https://www.nice.org.uk/about/what-we-do/nice-syndication-api",
    "text": "Syndicate our content"
  },
  {
    "href": "https://www.nice.org.uk/events",
    "text": "Events"
  }
]
