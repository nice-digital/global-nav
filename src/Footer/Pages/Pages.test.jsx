import Pages from "./Pages";
import { render } from "@testing-library/react";

describe("Pages", () => {
	it("Matches snapshot", () => {
		const { container } = render(<Pages />);
		expect(container).toMatchSnapshot();
	});

	// Skipping this test, as the custom pathways feedback URL is no longer used
	// If we start overriding links per service again then we can reinstate the test
	it.skip("Uses overridden feedback URL as href for site with custom feedback URL", () => {
		const { getByText } = render(<Pages service="pathways" />);
		expect(getByText("Leave feedback")).toHaveAttribute(
			"href",
			"https://www.nice.org.uk/NICE-Pathways-feedback"
		);
	});
});
