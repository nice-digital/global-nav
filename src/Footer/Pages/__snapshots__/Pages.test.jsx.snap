// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Matches snapshot 1`] = `
<div>
  <nav
    class="wrapper"
  >
    <ul
      class="list"
    >
      <li>
        <a
          href="https://www.nice.org.uk/about"
        >
          About NICE
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/about/what-we-do/into-practice"
        >
          Into practice
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/get-involved"
        >
          Get involved with NICE
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/get-involved/jobs"
        >
          Jobs
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/news"
        >
          News
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/news/nice-newsletters-and-alerts"
        >
          Newsletters and alerts
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/get-involved/contact-us"
        >
          Contact us
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/forms/leave-feedback"
        >
          Leave feedback
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/re-using-our-content"
        >
          Reusing our content
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/forms/use-of-nice-content-in-the-uk"
        >
          NICE UK Open Content Licence
        </a>
      </li>
      <li>
        <a
          href="https://www.nice.org.uk/about/what-we-do/nice-syndication-api"
        >
          Syndicate our content
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
